pandera/backends/pyspark/decorators.py

```diff
@@ -126,3 +131,54 @@ def _get_check_obj():
         return wrapper
 
     return _wrapper
+
+
+def cache_check_obj():
+    """This decorator evaluates if `check_obj` can be cached before validation.
+
+    As a new Spark action is triggered for every new data check added to the Pandera
+     schema by the user, Spark keeps reprocessing the `check_obj` dataframe lots of
+     times. To avoid such waste of processing resources and to decrease validation
+     times for complex scenarios, this decorator makes the `check_obj` be cached
+     before validation and unpersisted after it occurs.
+
+    The execution of this process depends on the `PANDERA_PYSPARK_CACHING` environment
+     variable.
+    """
+
+    def _wrapper(func):
+        @functools.wraps(func)
+        def wrapper(self, *args, **kwargs):
+
+            # Skip if not enabled
+            if CONFIG.pyspark_cache is not True:
+                return func(self, *args, **kwargs)
+
+            # Check if decorated function has a `check_obj` kwarg
+            try:
+                check_obj: DataFrame = kwargs["check_obj"]
```

Comment:
Here we are expecting that `check_obj` can only be specified as a kwarg, however to the definition it could also be an `arg`